### PR TITLE
Stm32f103xb interface

### PR DESCRIPTION
**Pull request by c1728p9**
Add support for the stm32f103 interface to DAPLink. Thanks, @zoujixing for doing most of the porting work in PR #133. This patch is based on yours but replaces the HAL with the stm32 cube driver so the license is open source. It also replaces the USB driver with an open source version. Finally, it is rebased and configured for NUCLEO boards.

### CODE REVIEW

**Comment by Awais174** [comment]
Thanks @c1728p9  for carrying out this work.
When do you intend to merge this pull request with master?

**Comment by c1728p9** [comment]
CI is down, so once it is working I'll run testing on this and then merge it.

**Comment by Awais174** [comment]
That will be great. 
If possible, could please also add support for target STM32F437VG (256KB).

**Comment by c1728p9** [comment]
Hi @Awais174, is there a board you had in mind for the STM32F437VG?

**Comment by Awais174** [comment]
@c1728p9 u-blox C030 board (either U2XX or N2XX) is based on STM32F437VG as target MCU and STM32F103 as interface MCU.

File: source/board/stm32f072rb.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @file    stm32f103rb.c
+ * @brief   board ID for the NXP NUCLEO-F103RB board
```
Comment (therealprof) [inline source/board/stm32f072rb.c]: This is stm32f072rb.c and there ain't no such thing as a NXP Nucleo board ...

File: source/board/stm32f103rb.c [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @file    stm32f103rb.c
+ * @brief   board ID for the NXP NUCLEO-F103RB board
+ *
```
Comment (therealprof) [inline source/board/stm32f103rb.c]: Same here.

File: source/board/stm32f207zg.c [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @file    stm32f103rb.c
+ * @brief   board ID for the NXP NUCLEO-F103RB board
+ *
```
Comment (therealprof) [inline source/board/stm32f207zg.c]: Same here, different file and no NXP Nucleo.

File: source/board/stm32f334r8.c [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @file    stm32f103rb.c
+ * @brief   board ID for the NXP NUCLEO-F103RB board
+ *
```
Comment (therealprof) [inline source/board/stm32f334r8.c]: Dito.

File: source/board/stm32f411re.c [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @file    stm32f103rb.c
+ * @brief   board ID for the NXP NUCLEO-F103RB board
+ *
```
Comment (therealprof) [inline source/board/stm32f411re.c]: Dito.

File: source/board/stm32f746zg.c [hunk 6]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @file    stm32f103rb.c
+ * @brief   board ID for the NXP NUCLEO-F103RB board
+ *
```
Comment (therealprof) [inline source/board/stm32f746zg.c]: And another time.

File: source/board/stm32l476rg.c [hunk 7]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @file    stm32f103rb.c
+ * @brief   board ID for the NXP NUCLEO-F103RB board
+ *
```
Comment (therealprof) [inline source/board/stm32l476rg.c]: And again.

**Comment by therealprof** [comment]
Any plans to pursuing this any further?

**Comment by c1728p9** [comment]
@therealprof yeah eventually, but this is a low priority right now.

**Comment by RobMeades** [comment]
<bump>.

**Comment by RobMeades** [comment]
Hi Russ.  We need to make changes for our C030 product (which uses the STM32F437VG as target MCU and STM32F103 as interface MCU) that rely on this PR being submitted.  Is there any way we can help to accelerate its submission?

**Comment by cederom** [comment]
I can help testing :-)

**Pull request merged**